TomTomSDKExamples/Examples/UseCase/Offline/OfflineNavigationContent.swift

```diff
@@ -0,0 +1,689 @@
+        if let offlineDataProvider = OfflineTileDataProviderFactory.createOfflineTileDataProvider(store: ndsStore) {
+            dataProviders.append(offlineDataProvider)
+        } else {
+            /* YOUR CODE GOES HERE */
+            fatalError("Error!")
+        }
+
+        /// Create the map options by creted data provider and other configurations
+        let mapOptions = MapOptions(
+            apiKey: Keys.apiKey,
+            cachePolicy: .noCaching,
+            styleMode: .main,
+            dataProviders: dataProviders
+        )
+
+        /// Create the "MapView" object with the map options
+        mapView = TomTomSDKMapDisplay.MapView(mapOptions: mapOptions)
+    }
+}
+
+// MARK: UIViewRepresentable
+
+/// Extend the TomTomOfflineMapView to conform to the UIViewRepresentable protocol.
+/// This allows you to use the MapView in SwiftUI
+extension TomTomOfflineMapView: UIViewRepresentable {
+    typealias UIViewType = TomTomSDKMapDisplay.MapView
+
+    func makeUIView(context: Context) -> TomTomSDKMapDisplay.MapView {
+        mapView.delegate = context.coordinator
+        mapView.currentLocationButtonVisibilityPolicy = .hiddenWhenCentered
+        mapView.compassButtonVisibilityPolicy = .visibleWhenNeeded
+        return mapView
+    }
+
+    func updateUIView(_: TomTomSDKMapDisplay.MapView, context: Context) {
+        updateEdgeInsets(context: context)
+    }
+
+    func makeCoordinator() -> OfflineMapCoordinator {
+        OfflineMapCoordinator(mapView, offlineNavigationController: offlineNavigationController)
+    }
+}
+
+// MARK: - OfflineMapCoordinator
+
+/// Create the OfflineMapCoordinator, which facilitates communication from the UIView to the SwiftUI environments.
+final class OfflineMapCoordinator: NSObject {
+    // MARK: Lifecycle
+
+    init(_ mapView: TomTomSDKMapDisplay.MapView, offlineNavigationController: OfflineNavigationController) {
+        self.offlineNavigationController = offlineNavigationController
+        self.mapView = mapView
+        super.init()
+        observe(offlineNavigationController: offlineNavigationController)
+    }
+
+    // MARK: Internal
+
+    func setContentInsets(edgeInsets: EdgeInsets) {
+        mapView.contentInsets = NSDirectionalEdgeInsets(
+            top: edgeInsets.top,
+            leading: edgeInsets.leading,
+            bottom: edgeInsets.bottom,
+            trailing: edgeInsets.trailing
+        )
+    }
+
+    // MARK: Private
+
+    private let offlineNavigationController: OfflineNavigationController
+    private let mapView: TomTomSDKMapDisplay.MapView
+    private var map: TomTomSDKMapDisplay.TomTomMap?
+    private var routeOnMap: TomTomSDKMapDisplay.Route?
+    private var cameraUpdated = false
+    private var cancellableBag = Set<AnyCancellable>()
+}
+
+// MARK: TomTomSDKMapDisplay.MapViewDelegate
+
+/// Extend the OfflineMapCoordinator to conform to the MapViewDelegate protocol by implementing the onMapReady callback.
```

Comment:
Not really useful, since duplicates the definition.

Reply:
👍 Done!